### ISSUE TITLE
Needless continue for match block

### ISSUE THREAD

**Issue by pickfire** [rust-lang/rust-clippy#4077]
tl;dr Fix simple needless continue

```rust
for _ in 0..1 {
    // do something
    continue;
}
```

Currently, `needless_continue` only checks possible removal of `continue` by reordering `if-else` block. One more simpler needless continue to be removed:

From https://github.com/Gymmasssorla/finshir/blob/eae5d0c6761f5558c8ed33ef098d8bd13a07e64f/src/testing/mod.rs#L124-L136

```rust
        for _ in 0..(failed_count.get() - 1) {
            match socket.write_all(portion) {
                Ok(_) => return SendPortionResult::Success,
                Err(err) => {
                    error!(
                        "Failed to send {} byte(s) >>> {}! Retrying the operation...",
                        helpers::cyan(portion.len()),
                        err
                    );
                    continue;
                }
            }
        }
```

Which `continue` can be dropped.

```rust
        for _ in 0..(failed_count.get() - 1) {
            match socket.write_all(portion) {
                Ok(_) => return SendPortionResult::Success,
                Err(err) => {
                    error!(
                        "Failed to send {} byte(s) >>> {}! Retrying the operation...",
                        helpers::cyan(portion.len()),
                        err
                    );
                }
            }
        }
```

One more in https://github.com/Gymmasssorla/finshir/blob/eae5d0c6761f5558c8ed33ef098d8bd13a07e64f/src/testing/mod.rs#L150-L164

What needs to be achieve here is to check if there are any `continue` at the end of each branches (including `if-else` and `match` blocks).

Possible code changes to clippy can probably be done in https://github.com/rust-lang/rust-clippy/blob/master/clippy_lints/src/needless_continue.rs#L374

**Comment by Xaeroxe** [comment]
https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=e1f764cbed2b7227df7a748aa6a2b2dd

Similarly, any continue, conditional or not, which is the last statement of the loop it is in, is useless and should be omitted.
